### PR TITLE
[Cache] Adding about marshallers

### PR DESCRIPTION
**Pull request by Nyholm**
This is related to #15050

It will add information about serialisation, encryption and compression.

### CODE REVIEW

**Comment by bastnic** [comment]
How to use them?
When to use what? Igbinary better than compression?

**Comment by Nyholm** [comment]
I think it depends on your data. Some types of data might be better with Igbinary. I have not tested any of them so I cannot really tell. 

> How to use them?

I'll add a code example.

File: components/cache.rst [hunk 1]
```diff
@@ -192,6 +192,25 @@ Now you can create, retrieve, update and delete items using this cache pool::
 
 For a list of all of the supported adapters, see :doc:`/components/cache/cache_pools`.
 
+Serializing Data
+----------------
+
+When an item is stored in the cache, it is serialised to a string. It a class implementing
```
Comment (apfelbox) [inline components/cache.rst]: > It a class

Some word here is missing

Comment (Nyholm) [reply components/cache.rst]: Thank you.

File: components/cache.rst [hunk 2]
```diff
@@ -192,6 +192,25 @@ Now you can create, retrieve, update and delete items using this cache pool::
 
 For a list of all of the supported adapters, see :doc:`/components/cache/cache_pools`.
 
+Serializing Data
+----------------
+
+When an item is stored in the cache, it is serialised to a string. It is a class
+implementing :class:`Symfony\\Component\\Cache\\Marshaller\\MarshallerInterface`
+that is responsible for serializing and unserializing. Or to be technically correct:
+to ``marshall()``and to ``unmarshall()``.
```
Comment (bastnic) [inline components/cache.rst]: miss a space before `and`?

Comment (Nyholm) [reply components/cache.rst]:
```suggestion
to ``marshall()`` and to ``unmarshall()``.
```

Thank you

File: components/cache.rst [hunk 3]
```diff
@@ -192,6 +192,25 @@ Now you can create, retrieve, update and delete items using this cache pool::
...
+The :class:`Symfony\\Component\\Cache\\Marshaller\\DefaultMarshaller` is using PHP's
+``serialize()`` or ``igbinary_serialize()`` if the Igbinary extension is installed.
+There are other marshallers that will encrypt or compress the data before storing it::
+
+    use Symfony\Component\Cache\Adapter\RedisAdapter;
+    use Symfony\Component\Cache\DefaultMarshaller;
+    use Symfony\Component\Cache\DeflateMarshaller;
+
+    $marshaller = new DeflateMarshaller(new DefaultMarshaller());
+    $cache = new RedisAdapter(new \Redis(), 'namespace', 0, $marshaller);
```
Comment (bastnic) [inline components/cache.rst:212]: maybe add the way to do it framework side?

Comment (Nyholm) [reply components/cache.rst:212]: Not in this docs. This is for the component only.

**Comment by javiereguiluz** [comment]
Tobias, thanks a lot for contributing this. All this "marshallers thing" was confusing to me, so this was lacking in the docs.

While merging I did some changes to add an introductory note explaining, very briefly, the differences between serializing and marshalling. Cheers!

**Pull request merged**